bip-musig2/Makefile

```
@@ -0,0 +1,5 @@
+.PHONY: check
+check: key_agg_vectors.json nonce_agg_vectors.json nonce_gen_vectors.json sig_agg_vectors.json sign_verify_vectors.json tweak_vectors.json reference.py gen_vectors_helper.py
+	mypy reference.py > /dev/null
```

nit:
```suggestion
	mypy --no-error-summary reference.py
```

Wouldn't make a shell script more sense? Usually a make invocation is supposed to "make" something (a target) but here this is simply running some tests.

---

Fixed and replaced with a shell script. I usually like being able to run `make check` instead of having to remember whether the shell script is called `tests.sh`, `test.sh`, `run-tests.sh`, etc. But in this case we can use POSIX shell and drop the make dependency.